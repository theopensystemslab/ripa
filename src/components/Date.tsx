import Box from "@material-ui/core/Box";
import Button from "@material-ui/core/Button";
import Grid from "@material-ui/core/Grid";
import Input from "@material-ui/core/Input";
import InputLabel from "@material-ui/core/InputLabel";
import { useFormik } from "formik";
import * as React from "react";
import FocusWithin from "react-focus-within";

import Question from "./Question";

interface IDate {
  title: string;
  name: string;
  type: string;
  options: string[];
  inputProps: ILimits;
  includeSubmit?: boolean;
}
interface ILimits {
  day: { min: number; max: number };
  month: { min: number; max: number };
  year: { min: number; max: number };
}
export const Date: React.FC<IDate> = ({
  title,
  type,
  name,
  options,
  inputProps,
  includeSubmit = false
}) => {
  const formik = useFormik({
    initialValues: {},
    onSubmit: values => {
      console.log(JSON.stringify(values, null, 2));
    }
  });
  return (
    <Box py={4}>
      <FocusWithin>
        {({ isFocused, getFocusProps }) => (
          <form onSubmit={formik.handleSubmit} {...getFocusProps()}>
            <Box pb={1}>
              <Question inFocus={isFocused}>{title}</Question>
            </Box>
            <Grid container spacing={1}>
              {options.map((el, index) => (
                <Grid item key={`${el}-${index}`}>
                  <InputLabel htmlFor={name}>{el}</InputLabel>
                  <Input
                    required
                    id={name}
                    onChange={formik.handleChange}
                    type={type}
                    name={`${name}-${el}`}
                    inputProps={inputProps[el]}
                  />
                </Grid>
              ))}
            </Grid>
            {includeSubmit && (
              <Box pt={3}>
                <Button type="submit" variant="contained" color="primary">
                  Save and Continue
                </Button>
              </Box>
            )}
          </form>
        )}
      </FocusWithin>
    </Box>
  );
};

export default Date;
