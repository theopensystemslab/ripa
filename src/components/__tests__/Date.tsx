import { fireEvent, render } from "@testing-library/react";
import * as React from "react";

import Date from "../Date";

describe("Date Component", () => {
  let title: string;
  let name: string;
  let type: string;
  let options: string[];
  let inputProps: {
    day: { min: number; max: number };
    month: { min: number; max: number };
    year: { min: number; max: number };
  };
  beforeEach(() => {
    title = " test title";
    name = "test name";
    type = "number";
    options = ["day", "month", "year"];
    inputProps = {
      day: {
        min: 1,
        max: 31
      },
      month: {
        min: 1,
        max: 12
      },
      year: {
        min: 1920,
        max: 2020
      }
    };
  });
  it("should render snapshot", () => {
    const { asFragment } = render(
      <Date
        title={title}
        name={name}
        type={type}
        options={options}
        inputProps={inputProps}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should not accept text", () => {
    const { getByLabelText } = render(
      <Date
        title={title}
        name={name}
        type={type}
        options={options}
        inputProps={inputProps}
      />
    );
    fireEvent.change(getByLabelText(/day/i), {
      target: { value: "test text" }
    });
    expect(getByLabelText(/day/i)).not.toHaveValue("test text");
    fireEvent.change(getByLabelText(/month/i), {
      target: { value: "test text" }
    });
    expect(getByLabelText(/month/i)).not.toHaveValue("test text");
    fireEvent.change(getByLabelText(/year/i), {
      target: { value: "test text" }
    });
    expect(getByLabelText(/year/i)).not.toHaveValue("test text");
  });
});
