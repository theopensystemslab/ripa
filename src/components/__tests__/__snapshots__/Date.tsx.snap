// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <form>
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <div
          class="makeStyles-root-4"
        >
          <div
            class="MuiTypography-root makeStyles-question-5 MuiTypography-h4"
          >
            <strong>
               test title
            </strong>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
          >
            day
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline"
            required=""
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              max="31"
              min="1"
              name="test name-day"
              required=""
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
          >
            month
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline"
            required=""
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              max="12"
              min="1"
              name="test name-month"
              required=""
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated"
          >
            year
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline"
            required=""
          >
            <input
              class="MuiInputBase-input MuiInput-input"
              max="2020"
              min="1920"
              name="test name-year"
              required=""
              type="number"
              value=""
            />
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
